common/lib/xmodule/xmodule/tests/test_library_sourced_block.py

```diff
@@ -0,0 +1,73 @@
+"""
+Tests for Source form Library XBlock
+"""
+from openedx.core.djangoapps.content_libraries.tests.base import ContentLibrariesRestApiTest
+from student.roles import CourseInstructorRole
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.tests import get_test_system
+from xmodule.x_module import AUTHOR_VIEW, STUDENT_VIEW
+
+
+class LibrarySourcedBlockTestCase(ContentLibrariesRestApiTest):
+    """
+    Tests for LibraryToolsService which interact with blockstore-based content libraries
+    """
+    def setUp(self):
+        super().setUp()
+        self.store = modulestore()
+
+    def test_block_views(self):
+        # Create a blockstore content library
+        library = self._create_library(slug="testlib1_preview", title="Test Library 1", description="Testing XBlocks")
+        # Add content to the library
+        html_block_id = self._add_block_to_library(library["id"], "html", "html_student_preview")["id"]
+        self._set_library_block_olx(html_block_id, '<html>Student Preview Test</html>')
+
+        # Create a modulestore course
+        course = CourseFactory.create(modulestore=self.store, user_id=self.user.id)
+        CourseInstructorRole(course.id).add_users(self.user)
+        # Add Source from library block to the course
+        source_block = ItemFactory.create(
+            category="library_sourced",
+            parent=course,
+            parent_location=course.location,
+            user_id=self.user.id,
+            modulestore=self.store
+        )
+        sourced_block_location = source_block.location
+
+        html = self.get_block_view(source_block, AUTHOR_VIEW)
+        # Check if author_view for empty block renders using the editor template
+        self.assertIn('library-sourced-block-author-view.html', html)
```

Comment:
I'm curious - how is the template filename getting included in the HTML?

Reply:
Mako in the test runtime is using [this mock function](https://github.com/edx/edx-platform/blob/0e2b612c1fb4f3e385f3004801aa5b5ed0221eda/common/lib/xmodule/xmodule/tests/__init__.py#L171-L178) to render the templates, which just returned the filename and context instead of rendering the template - which seems good enough for tests